tests/beta/test_default_tensor.py

```diff
@@ -1754,3 +1754,31 @@ def test_hermitian(self, theta, phi, varphi, monkeypatch, rep, tol):
             )
         ) / 16
         assert np.allclose(var, expected, atol=tol, rtol=0)
+
+
+class TestWiresIntegration:
+    """Test that the device integrates with PennyLane's wire management."""
+
+    def make_circuit_expval(self, wires):
+        """Factory for a qnode returning expvals using arbitrary wire labels."""
+        dev = qml.device("default.tensor", wires=wires)
+
+        @qml.qnode(dev)
+        def circuit():
+            qml.RX(0.5, wires=wires[0])
+            qml.RY(2., wires=wires[1])
+            qml.CNOT(wires=[wires[0], wires[2]])
+            qml.CNOT(wires=[wires[2], wires[1]])
+            return [qml.expval(qml.PauliZ(wires=w)) for w in wires]
+
+        return circuit
+
+    @pytest.mark.parametrize("wires1, wires2", [(['a', 'c', 'd'], [2, 3, 0]),
+                                                ([-1, -2, -3], ['q1', 'ancilla', 2])])
+    def test_wires_expval(self, wires1, wires2, tol):
+        """Test that the expectation of a circuit is independent from the wire labels used."""
+
+        circuit1 = self.make_circuit_expval(wires1)
+        circuit2 = self.make_circuit_expval(wires2)
+
+        assert np.allclose(circuit1(), circuit2(), tol)
```

Comment:
Nice! does this also work passing the wires as QNode arguments?

```python
@qml.qnode(dev)
def circuit(wires):
    qml.RX(0.5, wires=wires[0])
    qml.RY(2., wires=wires[1])
    qml.CNOT(wires=[wires[0], wires[2]])
    qml.CNOT(wires=[wires[2], wires[1]])
    return [qml.expval(qml.PauliZ(wires=w)) for w in wires]

wires =np.array(['a', 'c', 'd'], requires_grad=False)
circuit(wires)
```

Reply:
Seems to work. Full example:

```python
import pennylane as qml
from pennylane import numpy as np

dev = qml.device('default.qubit', wires=['a', 'c', 'd'])

@qml.qnode(dev)
def circuit(wires):
    qml.RX(0.5, wires=wires[0])
    qml.RY(2., wires=wires[1])
    qml.CNOT(wires=[wires[0], wires[2]])
    qml.CNOT(wires=[wires[2], wires[1]])
    return [qml.expval(qml.PauliZ(wires=w)) for w in wires]

wires =np.array(['a', 'c', 'd'], requires_grad=False)
print(circuit(wires))
>>> [ 0.87758256 -0.36520321  0.87758256]
```

Reply:
Should we use this in the tests? The reason I didn't is that the device also needs the correct wires...

Reply:
Sweet! I don't think it needs to be tested (probably it's already being tested by the QNode integration tests), was just curious